ussologin/visitwebpage.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package ussologin
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+
+	"github.com/juju/usso"
+	"gopkg.in/juju/environschema.v1/form"
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+
+	"github.com/juju/idmclient"
+)
+
+// VisitWebPage returns a function which will allow authentication with USSO
+// via the cli.
+// The user will be prompted for username, password and any two factor authentication
+// code via the command line. Existing oauth tokens can be obtained, or new ones stored
+// If non-nil, the given TokenStore is used to store the oauth token obtained during
+// the login process so that less interaction may be required in future.
```

Comment:
If UbuntuSSO OAuth login is not available then this function falls back to httpbakery.OpenWebBrowser. ?